net/Socket.cpp

```diff
@@ -651,7 +738,7 @@ void SocketPoll::takeSocket(const std::shared_ptr<SocketPoll> &fromPoll,
     ASSERT_CORRECT_THREAD();
 
     // hold a reference during transfer
-    std::shared_ptr<Socket> socket = inSocket;
+    std::shared_ptr<Socket> socket = inSocket; // NOLINT
```

Comment:
What issue is `NOLINT` fixing?

Reply:
Fixes a warning from `clang-tidy`, noting that a copy might not be needed here (performance, prefer reference).
However, since we probably pass this off into an async case - *and* I didn't not completely review the code, I marked it to silent clang-tidy. 

In case the `socket` reference gets *copied* into the async-callback lambda capture, the shared_ptr counter would be increased as required and we could just use a reference for above `socket`. But as I wrote, I didn't double-check yet.

Reply:
I'd suggest a different take. Let's make the change when/if we check and have an answer. Silencing a tool buries a potentially valid warning. What's the use of that if we haven't checked its validity one way or the other?

Silencing it *is* a decision, and implies someone checked and made a decision. But, if that's not the case, then this is misleading and will make it that much harder to remember and revisit this. Best to leave it alone until we have the time to investigate. Only then we can decide to silence the tool or fix the issue (if it's an issue).

Reply:
Earmark: Removed the `NOLINT`, however - another patch may remove the explicit copy as it is being copied in the lambda capture. For full replication, the method itself shall receive a copy of the shared_ptr.
But will not touch this here.